homeassistant/components/zha/config_flow.py

```diff
@@ -473,10 +473,62 @@ async def async_step_maybe_reset_old_radio(
             temp_radio_mgr.device_settings = config_entry.data[CONF_DEVICE]
             temp_radio_mgr.radio_type = RadioType[config_entry.data[CONF_RADIO_TYPE]]
 
-            await temp_radio_mgr.async_reset_adapter()
+            try:
+                await temp_radio_mgr.async_reset_adapter()
+            except HomeAssistantError:
+                # Old adapter not found or cannot connect, show prompt to plug back in
+                return await self.async_step_plug_in_old_radio()
 
         return await self.async_step_maybe_confirm_ezsp_restore()
 
+    async def async_step_plug_in_old_radio(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Prompt user to plug in the old radio if connection fails."""
+        config_entries = self.hass.config_entries.async_entries(
+            DOMAIN, include_ignore=False
+        )
+
+        # config entry should always exist here, skip if not
+        if not config_entries:
+            return await self.async_step_maybe_confirm_ezsp_restore()
```

Comment:
This is the one line missing test coverage. Can write a test for this, but we should never be able to hit this.

We could assert that `config_entries` is not empty, or we could also save the "old adapter device path" to some instance variable in `async_step_maybe_reset_old_radio`.
Passing it as an additional argument also works, but it's not a pattern I've seen used for any config flows really.

Reply:
You conceivably could hit this by deleting ZHA halfway through the flow but I agree, this isn't something we should care about. There are many places in Core that break if you try hard enough.

Reply:
Alternative to this would be saving or passing the old adapter's path from `async_step_maybe_reset_old_radio`. Then, we wouldn't need to get config entries in the new step displaying the old adapter's path, so we could avoid the potential step skip if the entry is removed in time, thus avoiding the whole tests for it as well.

But now, we have the test and everything functions as expected if the user is able to hit this 😄